File: frontend/src/interfaces/Beatmap.ts
interface Beatmap {
    id: number;
    beatmapset_id: number;
    difficulty_rating: number;
    version: string;
    accuracy: number;
    ar: number;
    bpm: number;
    cs: number;
    status: string;
    url: string;
    total_length: number;
    user_id: number;
    beatmap_stats: BeatmapStats;
    last_updated: string; // Assuming time.Time is serialized as string in JSON
}

interface BeatmapStats {
    [key: string]: BeatmapStatsModel;
}

interface BeatmapStatsModel {
    play_count: number;
    pass_count: number;
}

File: backend/internal/service/osuapi/models/beatmap.go
package osuapimodels

import "time"

type Beatmap struct {
	Id               int       `json:"id"`
	BeatmapsetId     int       `json:"beatmapset_id"`
	DifficultyRating float64   `json:"difficulty_rating"`
	Version          string    `json:"version"`
	Accuracy         float64   `json:"accuracy"`
	Ar               float64   `json:"ar"`
	Bpm              float64   `json:"bpm"`
	Cs               float64   `json:"cs"`
	Status           string    `json:"status"`
	Url              string    `json:"url"`
	TotalLength      int       `json:"total_length"`
	UserId           int       `json:"user_id"`
	Passcount        int       `json:"passcount"`
	Playcount        int       `json:"playcount"`
	LastUpdated      time.Time `json:"last_updated"`
}
